caravel/model/listing.py

```diff
@@ -0,0 +1,54 @@
+from google.appengine.ext import ndb, db
+from caravel.utils import Principal, Device
+from caravel.model.moderation import ModeratedMixin
+from caravel.model.temporal import TimeOrderMixin
+from caravel.model.categories import CategoriesMixin
+from caravel.model.attachments import PhotosMixin
+from caravel.model.migration import SchemaMixin
+from caravel.model.priced import PriceMixin
+from caravel.model.principal import PrincipalMixin
+from caravel.model.side_effects import SideEffectsMixin
+from caravel import utils
+
+import datetime
+import re
+from flask import render_template
+
+class _Listing(CategoriesMixin, PhotosMixin, PrincipalMixin, TimeOrderMixin,
+               SchemaMixin, PriceMixin, ModeratedMixin, ndb.Model):
+
+    SCHEMA_VERSION = 11
+
+    title = ndb.StringProperty()
+    body = ndb.StringProperty()
+
+class Listing(SideEffectsMixin, _Listing):
+    def side_effects(self):
```

Comment:
Well, I want to decouple the `side_effects` hook from what the side effects actually do — this could include anything from an email to a webhook.